File: src/main/java/nl/chimpgamer/networkmanager/extensions/autoban/AutoBan.java
package nl.chimpgamer.networkmanager.extensions.autoban;

import lombok.Getter;
import nl.chimpgamer.networkmanager.api.extensions.NMExtension;
import nl.chimpgamer.networkmanager.api.utils.PlatformType;
import nl.chimpgamer.networkmanager.extensions.autoban.configuration.Settings;
import nl.chimpgamer.networkmanager.extensions.autoban.listeners.PunishmentListener;

@Getter
public class AutoBan extends NMExtension {
    private Settings settings;

    @Override
    protected void onEnable() {
        if (this.getNetworkManager().getPlatformType() != PlatformType.BUNGEECORD) {
            this.getLogger().severe("Hey, this NetworkManager extension is for BungeeCord only!");
            return;
        }
        this.settings = new Settings(this);

        this.getNetworkManager().registerListener(new PunishmentListener(this));
    }

    @Override
    protected void onDisable() {

    }

    @Override
    protected void onConfigsReload() {
        this.getSettings().reload();
    }
}

File: src/main/java/nl/chimpgamer/networkmanager/extensions/autoban/AutoBan.kt
package nl.chimpgamer.networkmanager.extensions.autoban

import nl.chimpgamer.networkmanager.api.extensions.NMExtension
import nl.chimpgamer.networkmanager.api.utils.PlatformType
import nl.chimpgamer.networkmanager.extensions.autoban.configuration.Settings
import nl.chimpgamer.networkmanager.extensions.autoban.listeners.PunishmentListener

class AutoBan : NMExtension() {
    val settings = Settings(this)
    lateinit var punishmentListener: PunishmentListener

    override fun onEnable() {
        if (networkManager.platformType !== PlatformType.BUNGEECORD) {
            logger.severe("Hey, this NetworkManager extension is for BungeeCord only!")
            return
        }
        settings.load()
        punishmentListener = PunishmentListener(this)
        networkManager.eventHandler.registerListener(punishmentListener)
    }

    override fun onDisable() {
        networkManager.eventHandler.unregisterListener(punishmentListener)
    }

    override fun onConfigsReload() {
        settings.reload()
    }
}